<?xml version="1.0"?>

<launch>
		<include file="$(find svo_ros)/launch/live.launch"/>

  	  	<node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
    		<param name="output_frame" value="odom"/>
    		<param name="freq" value="30.0"/>
    		<param name="sensor_timeout" value="1.0"/>
    		<param name="odom_used" value="false"/>
    		<param name="imu_used" value="true"/>
    		<param name="vo_used" value="false"/>
    		<param name="debug" value="false"/>
    		<param name="self_diagnose" value="false"/>

    		<remap from="imu_data" to="/hardcom/InertialData" />
    		<remap from="vo" to="/pose2odom/EstimatedPose" />

  		</node>

  		<node pkg="airship_keyboard" type="pose2odom.py" name="pose2odom"  output="screen">
        	<param name="rate" value="30.0" type="double"/>
			<param name="topic" value="pose2odom" type="string"/>
        </node>

        <node pkg="airship_keyboard" type="hardcom.py" name="hardcom_runalone"  output="screen">
        	<param name="rate" value="30.0" type="double"/>
			<param name="topic" value="hardcom" type="string"/>
        </node>

		<!-- <node pkg="airship_keyboard" type="odometry.py" name="odometry_runalone"  output="screen">
        	<param name="rate" value="30.0" type="double"/>
			<param name="topic" value="odometry" type="string"/>
			<param name="topic_hardcom" value="hardcom" type="string"/>
			<param name="topic_camcom" value="cameracom" type="string"/>
			<param name="topic_command" value="command" type="string"/>
        </node> -->

		<node pkg="airship_keyboard" type="cameracom.py" name="cameracom_runalone"  output="screen">
			<param name="rate" value="30.0" type="double"/>
			<param name="topic" value="cameracom" type="string"/>
			<param name="topic_command" value="command" type="string"/>
		</node>

		<node pkg="airship_keyboard" type="trajectory_planner.py" name="trajectory_planner_runalone"  output="screen">
			<param name="rate" value="30.0" type="double"/>
			<param name="topic" value="trajectory_planner" type="string"/>
			<param name="topic_hardcom" value="hardcom" type="string"/>
			<param name="topic_camcom" value="cameracom" type="string"/>
			<param name="topic_command" value="command" type="string"/>
			<param name="topic_odometry" value="odometry" type="string"/>
		</node>

        <node pkg="airship_keyboard" type="command.py" name="command"  output="screen">
        	<param name="rate" value="30.0" type="double"/>
			<param name="topic" value="command" type="string"/>
			<param name="topic_hardcom" value="hardcom" type="string"/>
			<param name="topic_trajectory_planner" value="trajectory_planner" type="string"/>
			<param name="topic_odometry" value="odometry" type="string"/>
		</node>
</launch>
